src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisTryWithResourcesOnlyOverlappingFalse.java

```diff
@@ -0,0 +1,117 @@
+/*
+RequireThis
+checkFields = (default)true
+checkMethods = (default)true
+validateOnlyOverlapping = false
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.coding.requirethis;
+
+import java.io.*;
+import java.util.Scanner;
+
+public class InputRequireThisTryWithResourcesOnlyOverlappingFalse implements AutoCloseable{
+    private BufferedReader br;
+    private InputStreamReader isr;
+    private Scanner sc;
+
+    void oneResource() {
+        try (BufferedReader br
+                     = new BufferedReader(new InputStreamReader(null, "utf-8"))) { } // ok
+        catch (IOException e) { }
+    }
+
+    void twoResourcesReferencingEachOther() {
+        try (InputStreamReader isr = new InputStreamReader(null, "utf-8"); // ok
+             BufferedReader br = new BufferedReader(isr)) { } // ok
+        catch (IOException e) { }
+    }
+
+    void threeResourcesReferencingEachOther() {
+        try (InputStreamReader isr = new InputStreamReader(null, "utf-8"); // ok
+             BufferedReader br = new BufferedReader(isr); // ok
+             Scanner sc = new Scanner(isr.toString() + br.toString())) { } // ok
+        catch (IOException e) { }
+    }
+
+    private String charset = "utf-8";
```

Comment:
It took me a while to find this field.
Is it required to have it here? Outside of other Fields

Reply:
Can we name all fields with prefix "field" to be more explicit

Reply:
> with prefix "field"

No reason apart from usage really, will move it back with the other fields on the top.

Reply:
Oops. Forgot this one in one of the files. Done.

Reply:
CI failure looks unrelated, yes?